packages/expo-audio/src/index.ts

```diff
@@ -33,6 +33,20 @@ export function useAudioPlayerStatus(player: AudioPlayer): AudioStatus {
   return useEvent(player, 'onPlaybackStatusUpdate', currentStatus);
 }
 
+export function useAudioSampleListener(
+  player: AudioPlayer,
+  listener: (data: { channels: { frames: number[] }[]; timestamp: number }) => void
+) {
+  player.setAudioSamplingEnabled(true);
+  useEffect(() => {
+    const subscription = player.addListener('onAudioSampleUpdate', listener);
```

Comment:
```suggestion
  useEffect(() => {
    const subscription = player.addListener('onAudioSampleUpdate', listener);
    player.setAudioSamplingEnabled(true);
```

maybe we should call `setAudioSamplingEnabled()` inside useEffect hook. otherwise it will called in every rendering loop